### PR TITLE
Categories index command

### PR DESCRIPTION
**Pull request by Jade-GG**
Because this is getting re-used all over the place.

### CODE REVIEW

**Comment by royduin** [comment]
Thanks! Looking forward to the implementation in the autocomplete 😇

File: src/Commands/IndexCategoriesCommand.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    /**
+     * Execute the console command.
+     */
+    public function handle(): int
+    {
+        $storeModel = config('rapidez.models.store');
+        $categoryModel = config('rapidez.models.category');
+        $stores = $this->argument('store')
+        ? $storeModel::where('store_id', $this->argument('store'))->get()
+        : $storeModel::all();
```
Comment (royduin) [inline src/Commands/IndexCategoriesCommand.php:34]:
```suggestion
        $stores = $this->argument('store')
            ? $storeModel::where('store_id', $this->argument('store'))->get()
            : $storeModel::all();
```

File: src/Commands/IndexCategoriesCommand.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    private function setStore(Store $store): void
+    {
+        config()->set('rapidez.store', $store->store_id);
+        $code = config('rapidez.models.store')::getCachedWhere(function ($store) {
+            return $store['store_id'] == config('rapidez.store');
+        })['code'];
+        config()->set('rapidez.store_code', $code);
+    }
```
Comment (royduin) [inline src/Commands/IndexCategoriesCommand.php:76]: Can we extract this somewhere so we can benefit from this everywhere? Could also be used within the products index command.

File: src/Commands/IndexCategoriesCommand.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        $stores = $this->argument('store')
+        ? $storeModel::where('store_id', $this->argument('store'))->get()
+        : $storeModel::all();
+
+        foreach ($stores as $store) {
+            $this->line('Indexing categories for store: '.$store->name);
+            $this->setStore($store);
+
+            $categoryModelInstance = new $categoryModel();
+            $data = config('rapidez.models.category')::select($categoryModelInstance->qualifyColumns(['entity_id', 'name', 'url_path', 'children_count']))
```
Comment (royduin) [inline src/Commands/IndexCategoriesCommand.php:41]: This is pretty static now, maybe add the possibility to change/extend this with for example an Eventy filter

File: src/Commands/IndexCategoriesCommand.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                ->whereNotNull('url_key')->whereNot('url_key', 'default-category')
+                ->where('children_count', '>', 0)
```
Comment (royduin) [inline src/Commands/IndexCategoriesCommand.php:43]: So categories without child categories won't be indexed? Why?

**Review by royduin — CHANGES_REQUESTED** [review]

**Comment by Jade-GG** [comment]
waiting for https://github.com/rapidez/core/pull/254 for now :)

**Comment by royduin** [comment]
#254 is merged 🚀

**Comment by Jade-GG** [comment]
https://github.com/rapidez/core/pull/280 😇

**Pull request closed without merging**